### PR TITLE
Basic XR Testing support

### PR DESCRIPTION
**Pull request by Manishearth**
This adds support for the XRTest and FakeXRDeviceController APIs from https://github.com/immersive-web/webxr-test-api, and plugs them into the `mock` backend of rust-webvr.

Tested with [a modified webxr test page](https://github.com/immersive-web/webxr-test-api)

r? @jdm @asajeffrey

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23485)

### CODE REVIEW

File: components/config/prefs.rs [hunk 1]
```diff
@@ -294,6 +294,8 @@ mod gen {
                 webxr: {
                     #[serde(default)]
                     enabled: bool,
+                    #[serde(default)]
+                    test: bool,
```
Comment (asajeffrey) [inline components/config/prefs.rs]: A bit annoying that this does something different than `dom.webvr.test` which renders to a GLWindow. At some point we'll have to come up with consistent naming.

File: components/script/dom/fakexrdevicecontroller.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            .as_window()
+            .webvr_thread()
+            .unwrap()
+            .send(WebVRMsg::MessageMockDisplay(msg))
+            .unwrap();
+    }
+}
+
+impl FakeXRDeviceControllerMethods for FakeXRDeviceController {
+    // check-tidy: no specs after this line
```
Comment (asajeffrey) [inline components/script/dom/fakexrdevicecontroller.rs]: No spec links :(

**Review by asajeffrey — APPROVED** [review]
LGTM with minor nits.

File: components/script/dom/fakexrdevicecontroller.rs (hunk 2 above)
Comment (Manishearth) [reply components/script/dom/fakexrdevicecontroller.rs]: there's no spec! there's https://github.com/immersive-web/webxr-test-api , but it's not really a spec. I've linked to it in a bunch of places

File: components/config/prefs.rs (hunk 1 above)
Comment (Manishearth) [reply components/config/prefs.rs]: I'm also okay with this rendering to a GLWindow in the future, but I'd have to teach the mock backend about it

**Pull request merged**